apps/api/src/app/subscribers-v2/channel-connections.controller.ts

```diff
@@ -0,0 +1,201 @@
+  })
+  @ApiResponse(GetChannelConnectionResponseDto, 200)
+  @RequirePermissions(PermissionsEnum.SUBSCRIBER_READ)
+  @RequireAuthentication()
+  async getChannelConnections(
+    @UserSession() user: UserSessionData,
+    @Param('subscriberId') subscriberId: string,
+    @Query() query: GetChannelConnectionsQueryDto
+  ): Promise<GetChannelConnectionResponseDto[]> {
+    await this.checkFeatureEnabled(user);
+
+    return await this.getChannelConnectionsUsecase.execute(
+      GetChannelConnectionsCommand.create({
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        resource: makeResourceKey(RESOURCE.SUBSCRIBER, subscriberId),
+        channel: query.channel,
+        provider: query.provider,
+      })
+    );
+  }
+
+  @Post('/:subscriberId/channel-connections')
+  @ApiOperation({
+    summary: 'Create channel connection for a subscriber for given integration',
+    description: `Create a new channel connection for a subscriber for given integration.`,
+  })
+  @ApiResponse(GetChannelConnectionResponseDto, 201)
+  @RequirePermissions(PermissionsEnum.SUBSCRIBER_WRITE)
+  @RequireAuthentication()
+  async createChannelConnection(
+    @UserSession() user: UserSessionData,
+    @Param('subscriberId') subscriberId: string,
+    @Body() body: CreateChannelConnectionRequestDto
+  ): Promise<GetChannelConnectionResponseDto> {
+    await this.checkFeatureEnabled(user);
+
+    return await this.createChannelConnectionUsecase.execute(
+      CreateChannelConnectionCommand.create({
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        identifier: body.identifier,
+        integrationIdentifier: body.integrationIdentifier,
+        resource: makeResourceKey(RESOURCE.SUBSCRIBER, subscriberId),
+        workspace: body.workspace,
+        auth: body.auth,
+      })
+    );
+  }
+
+  @Patch('/:identifier')
+  @ApiOperation({
+    summary: 'Update channel connection',
+    description: `Update an existing channel connection by its unique identifier.`,
+  })
+  @ApiParam({ name: 'identifier', description: 'The unique identifier of the channel connection', type: String })
+  @ApiResponse(GetChannelConnectionResponseDto, 200)
+  @RequirePermissions(PermissionsEnum.SUBSCRIBER_WRITE)
+  @RequireAuthentication()
+  async updateChannelConnection(
+    @UserSession() user: UserSessionData,
+    @Param('identifier') identifier: string,
+    @Body() body: UpdateChannelConnectionRequestDto
+  ): Promise<GetChannelConnectionResponseDto> {
+    await this.checkFeatureEnabled(user);
+
+    return await this.updateChannelConnectionUsecase.execute(
+      UpdateChannelConnectionCommand.create({
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        identifier,
+        workspace: body.workspace,
+        auth: body.auth,
+      })
+    );
+  }
+
+  @Delete('/:identifier')
```

Comment:
Should we prefix this and the top one with channel connections? As currently it's directly under the subscriber path